pkg/util/proxy.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"os"
+	"strings"
+)
+
+var defaultNoProxies = []string{"localhost", "127.0.0.1", "172.30.1.1"}
+
+// ProxyConfig keeps the proxy configuration for the current environment
+type ProxyConfig struct {
+	httpProxy  string
+	httpsProxy string
+	noProxy    []string
+}
+
+// NewProxyConfig creates a proxy configuration with the specified parameters
+func NewProxyConfig(httpProxy string, httpsProxy string, noProxy string) (*ProxyConfig, error) {
+	err := validateProxyURL(httpProxy)
+	if err != nil {
+		return nil, err
+	}
+
+	err = validateProxyURL(httpsProxy)
+	if err != nil {
+		return nil, err
+	}
+
+	np := []string{}
+	np = append(np, defaultNoProxies...)
+	if noProxy != "" {
+		np = append(np, strings.Split(noProxy, ",")...)
+	}
+	config := ProxyConfig{
+		httpProxy:  httpProxy,
+		httpsProxy: httpsProxy,
+		noProxy:    np,
+	}
+
+	return &config, nil
+}
+
+func (p *ProxyConfig) ProxyConfig() []string {
+	config := []string{}
+	if p.httpProxy != "" {
+		config = append(config, fmt.Sprintf("HTTP_PROXY=%s", p.httpProxy))
+	}
+
+	if p.httpsProxy != "" {
+		config = append(config, fmt.Sprintf("HTTPS_PROXY=%s", p.httpsProxy))
+	}
+
+	if p.httpProxy != "" || p.httpsProxy != "" {
+		config = append(config, fmt.Sprintf("NO_PROXY=%s", p.NoProxy()))
+	}
+
+	return config
+}
+
+func (p *ProxyConfig) HttpProxy() string {
+	return p.httpProxy
+}
+
+func (p *ProxyConfig) HttpsProxy() string {
+	return p.httpsProxy
+}
+
+func (p *ProxyConfig) NoProxy() string {
+	return strings.Join(p.noProxy, ",")
+}
+
+func (p *ProxyConfig) AddNoProxy(exclude string) {
+	p.noProxy = append(p.noProxy, exclude)
+}
+
+func (p *ProxyConfig) SetToEnvironment() {
```

Comment:
@hferentschik this is set to the process itself, and not to the parent shell... should this somehow be mentioned in the function name? `SetEnvironmentVariablesForProcess`... since I want to document somehow that this function does NOT have the side-effect of polluting the shell environment that started the minishift command.